### ISSUE TITLE
Problem loading fonts

### ISSUE THREAD

**Issue by pipriles** [jwilm/alacritty#582]
Hello, I am trying to use this font: [ypn envypn](https://ywstd.fr/p/pj/#envypn)

My configuration looks like this:

```yml
font:
  normal:
    family: ypn envypn 
  bold:
    family: ypn envypn 
  italic:
    family: ypn envypn 
  size: 9.0
  offset:
    x: 0.0
    y: 0.0
  use_thin_strokes: false
```

But i get the following error:

```bash
thread 'main' panicked at 'index out of bounds: the len is 0 but the index is 0', src/libcollections/vec.rs:1431
note: Run with `RUST_BACKTRACE=1` for a backtrace.
```

Did i miss something? I tried to adjust the size and the offset to see if there would be any difference but it's the same, is this a bug?

**Comment by jwilm** [comment]
Hmm, would you mind running this again with `RUST_BACKTRACE=1 alacritty`? Hard to say what the problem is with the short error.

**Comment by DarkDefender** [comment]
I think that this might be fixed with #590. At least for me, I got the same error when I tried to use bitmap fonts before that was merged. But now it seems to be working nicely :)

**Comment by pipriles** [comment]
Sorry for the delay..
Ok, i recompiled alacritty and now works if i set the font size to 10.0. But the font which alacritty displays is the version with the pixelsize set to 13, I tried to put the pixelsize to 15, but i don't know how.

I tried with a lot of numbers from 10 to 20 and i can't display the correct terminal font size.
```sh
Alacritty encountered an unrecoverable error:

	Invalid pixel size


```
Thank you for the response.

**Comment by DarkDefender** [comment]
Yeah, I also had that problem. It took a while for me to figure out which size the font wanted.

@jwilm In most other programs I used, it either auto adjusts the size to the nearest correct one, or it prints out the available font sizes. Perhaps alacritty could do something like this?

I would be really happy if it for example printed out this instead:
```
Alacritty encountered an unrecoverable error:

	Invalid pixel size: 10. Available pixel sizes: 11, 15
```

**Comment by ojab** [comment]
With [bitmap] `xos4 Terminus` font the only working `size` in range 1.0-50.0 is `9.0` and alacritty says:
```
$ ./target/release/alacritty
got font path="/usr/share/fonts/terminus/ter-x12n.pcf.gz"
```
and it only works if dpi is default 96x96. If I set dpi to 48x48 or 192x192, for example, or change size to any other value, alacritty says
```
$ ./target/release/alacritty
got font path="/usr/share/fonts/terminus/ter-x12n.pcf.gz"
Alacritty encountered an unrecoverable error:

        Invalid pixel size
```
So I assume the issue is not really about uninformative error message, but about really strange font matching (or https://github.com/jwilm/alacritty/issues/619 should be reopened, if it's a separate issue).

There are multiple font sizes installed here:
```
]$ fc-list | grep xos4 | grep Regula | sort
/usr/share/fonts/terminus/ter-x12n.pcf.gz: xos4 Terminus:style=Regular
/usr/share/fonts/terminus/ter-x14n.pcf.gz: xos4 Terminus:style=Regular
/usr/share/fonts/terminus/ter-x16n.pcf.gz: xos4 Terminus:style=Regular
/usr/share/fonts/terminus/ter-x18n.pcf.gz: xos4 Terminus:style=Regular
/usr/share/fonts/terminus/ter-x20n.pcf.gz: xos4 Terminus:style=Regular
/usr/share/fonts/terminus/ter-x22n.pcf.gz: xos4 Terminus:style=Regular
/usr/share/fonts/terminus/ter-x24n.pcf.gz: xos4 Terminus:style=Regular
/usr/share/fonts/terminus/ter-x28n.pcf.gz: xos4 Terminus:style=Regular
/usr/share/fonts/terminus/ter-x32n.pcf.gz: xos4 Terminus:style=Regular
```
so I don't really understand why size 12 is always used, why it can be used with `size=9.0` (not 12.0) and why dpi setting affects this at all.

**Comment by NickHu** [comment]
I'm also experiencing this issue with Terminus, but no sizes seem to work for me.

**Comment by dbirks** [comment]
I can confirm that `xos4 Terminus` is working for me with `size: 9.0`.

**Comment by pipriles** [comment]
Still having the same issue with `ypn envypn` font.

**Comment by da-x** [comment]
Running with the environment variable `FC_DEBUG=16777215` prints a lot of info from `fontconfig`. 
BTW, tried freetype2's FT2_DEBUG (i.e. FT2_DEBUG="any:10") but it did not yield anything.

I think that the first issue is that `FcFontMatch` in fontconfig does not pick the right PCF file. When configuring alacritty for the Terminus font with size 32.0, `/usr/share/fonts/terminus/ter-112n.pcf.gz` gets picked (size 12). This, regardless of the font size that is passed to the match. I have lots of other `ter*.pcf.gz` files for other sizes, of course, and fontconfig prints _something_ about them.

The second issue is the font size disagreement inside freetype2 or around alacritty's use of it - I am not sure, have not delved into these pieces of code too much.

**Comment by Bl4ckb0ne** [comment]
[Tamzen](https://github.com/sunaku/tamzen-font) used with `Tamzen for Powerline` works with `size: 9.0` but nothing else

**Comment by da-x** [comment]
I've managed to fix this problem in pull request #776, more testers are welcomed.
